### PR TITLE
Support onPageView event on LatteDelegate

### PR DESCRIPTION
**Pull request by carmenlau**

### CODE REVIEW

File: sdk/src/main/java/com/oursky/authgear/latte/Latte.kt [hunk 1]
```diff
@@ -42,6 +44,7 @@ class Latte(
         return suspendCoroutine { k ->
             val app = authgear.core.application
             val broadcastAction = makeRandomAction(app)
+            val that = this
```
Comment (kiootic) [inline sdk/src/main/java/com/oursky/authgear/latte/Latte.kt:47]: Unused.

File: sdk/src/main/java/com/oursky/authgear/latte/WebViewEvent.kt [hunk 2]
```diff
@@ -3,3 +3,5 @@ package com.oursky.authgear.latte
 internal sealed class WebViewEvent {
     object OpenEmailClient : WebViewEvent()
 }
+
+internal class ViewPageEvent(val path: String) : WebViewEvent()
```
Comment (kiootic) [inline sdk/src/main/java/com/oursky/authgear/latte/WebViewEvent.kt:7]:
```kotlin
internal sealed class WebViewEvent {
    object OpenEmailClient : WebViewEvent()
    data class ViewPage(val path: String) : WebViewEvent()
}
```

File: sdk/src/main/java/com/oursky/authgear/latte/WebViewJSInterface.kt [hunk 3]
```diff
@@ -33,6 +34,10 @@ internal class WebViewJSInterface(private val webView: WebView) {
 
         when (type) {
             BuiltInEvent.OPEN_EMAIL_CLIENT -> this.webView.listener?.onEvent(WebViewEvent.OpenEmailClient)
+            BuiltInEvent.VIEW_PAGE -> {
+                val path = event.jsonObject["path"]?.jsonPrimitive?.contentOrNull ?: return
+                this.webView.listener?.onEvent(ViewPageEvent(path))
+            }
```
Comment (kiootic) [inline sdk/src/main/java/com/oursky/authgear/latte/WebViewJSInterface.kt:40]: Should probably wrap in a try-catch in case of unexpected event shape.

**Comment by carmenlau** [comment]
Updated to wrap the path with data class `LatteViewPageEvent` in 3532cbf, so that we can enrich the callback more easily in the future.

ref: https://github.com/authgear/authgear-sdk-ios/pull/113#discussion_r1119603613

**Pull request merged**